cumulus/xcm/xcm-emulator/src/lib.rs

```diff
@@ -1326,9 +1329,9 @@ pub struct TestArgs {
 }
 
 /// Auxiliar struct to help creating a new `Test` instance
-pub struct TestContext<T> {
-	pub sender: AccountId,
-	pub receiver: AccountId,
+pub struct TestContext<T, R: Chain, D: Chain> {
```

Comment:
what about aligning this R/D chain types?

this is little bit confusing:
sender -> **R** -> Origin + `Origin::RuntimeOrigin` looks not nice
**r**eceiver -> D -> Destination

I would suggest

`R: Chain` -> `SourceChain: Chain`
`D: Chain` -> `TargetChain: Chain`

and bellow:
Origin -> `SourceChain`
Destination -> `TargetChain`

Reply:
Renamed `R` and `D` to be aligned with the rest of the generics. I'll keep `Origin` and `Destination` names though as they are also descriptive.

Reply:
@NachoPal
ok, no problem :)

hrmp uses similar sender/recipient notation [here](https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/parachain/src/primitives.rs#L300-L304), but I guess receiver is also ok